### ISSUE TITLE
[FEATURE_REQUEST] Suggestions for Data Bank and RAG

### ISSUE THREAD

**Issue by lannashelton** [SillyTavern/SillyTavern#2296]
### Have you searched for similar requests?

Yes

### Is your feature request related to a problem? If so, please describe.

I'm playing with the new data bank and RAG feature since the 1.12.0 release came out and I've found some ways where it can be improved.

#### Batch Select Data Bank files
Lets say I've added hundreds of files into databank. And I want to delete all of them now. There should be a way to bulk select all files and delete them, similar to "bulk edit characters" feature.

#### RAG World Info
I think World Info entries should have a toggle button that turns them into separate vectors/RAG entries when turned on. Vectors are much better way to handle multiple little entries like World Info so I think it would be very useful to integrate them with RAG.

#### Make markdown clean-up optional
I've noticed that data bank cleans up markdown format when you add/scrape text. Sometimes it is more beneficial to keep those format intact. So I think there should be a toggle feature that says something like "Turn into plain text before adding to data bank" (yes/no)

#### Use a better embedding model for default
Default embedding model have low accuracy score and I think SillyTavern should use a better embedding model for default such as mixedbread-ai/mxbai-embed-large-v1

#### Add slash commands for data bank
You should add slash commands to add entries into data bank. I want to be able to create quick reply scripts that automatically store certain AI replies in data bank

### Priority

Medium (Would be very useful)

### Are you willing to test this on staging/unstable branch if this is implemented?

None

**Comment by Wolfsblvt** [comment]
I'll start numbering your suggestions, so context is better when replying.

1. **Batch Select Data Bank files**
  Sure, good idea. But bulk is a bit of tedious work to implement.
2. **RAG World Info**
  This already exists, and is pretty easy to use. Enable "Enabled for World Info" in the vector storage extension, then use the "Vectorized" status for all WI you want to have matched via vectors.
3. **Make markdown clean-up optional**
  Fixed in staging already. Markdown will not stripped, but preserved as is. Convert the file manually if you want it to. 
   https://github.com/SillyTavern/SillyTavern/commit/edf981a5a158f160082e45d4d0e06f297bda5b3f
4. **Use a better embedding model for default**
  Pretty sure you are free to choose whatever compatible embedding model you want to. See [documentation](https://docs.sillytavern.app/usage/core-concepts/data-bank/#local) for details. If you want to suggest another model as the default one, that might be something to look into. Though likely better to be discussed on the Discord. Or try pinging Cohee.
5. **Add slash commands for data bank**
  Not a bad idea. Though I am not sure if this is really needed. Slash commands for WI are there, and you can just build your on vector-enabled lorebook where you can add and update the entries. That should work like you want it to.  
  The data bank itself is more for actual files that you want to provide.

#### WI option to turn on vector lookup
![image](https://github.com/SillyTavern/SillyTavern/assets/9962104/ec9aec5a-b4b9-43b4-8cec-c259c4678656)

**Comment by lannashelton** [comment]
I saw that option in the World Info and tried it. From what I understand we are still required to write keywords ourselves. That's not exactly what I want. If you turn it on, fill the World Info entry but don't write keywords, then it doesn't work. What I wanted was to eliminate the need of writing trigger words manually. Because from a simple entry, embedding model can create dozens of vector links. I could never write that much trigger words manually. What I want is something like a "Vectorize All" button that creates vectors out of a World Info.

Edit: I've tried it again and it worked now. So yes, we can say that already exists. Thank you.
